File: CA4_calculator.py
# calculator.py
# calculate the factorial of a number.

#-----------------------------------------------
#        1- ADDITION
#-----------------------------------------------
	
def add (first,second) :
	return first + second
	
first = int(raw_input("enter first number "))	
second = int(raw_input("enter second number "))
print first + second	
#-----------------------------------------------
#       2- Subtraction
#-----------------------------------------------

def subtract (first,second):
	return first - second
	
#-----------------------------------------------
#       3- MULTIPLICATION
#-----------------------------------------------	
def multiply (first,second):
	return first * second		
#-----------------------------------------------
#        4- DIVISION
#-----------------------------------------------
	
def divide (first,second):
	if second == 0:
		return 'inf'
	else:
		return first / second
#-----------------------------------------------
#         5- EXPONENT
#-----------------------------------------------	
	
def exponent (first,second):
		
	if first != 0 and second ==1:
		return 1
	if first == 0 and second ==0:
		return "inf"
	if first == 0 and second >0:
		return 0
	if first != 0 and second ==0:
		return 1
	else:
		return first ** second	
#-----------------------------------------------
#         6- SQUARE
#-----------------------------------------------	
	
def square (first,second):
	if first == 0:
		return 0
	elif second == 2:
		return first ** second	

#-----------------------------------------------
#         7- CUBE
#-----------------------------------------------	
	
def cube (first,second):
	if first == 0:
		return 0
	elif second == 3:
		return first ** second	
		
#-----------------------------------------------
#       8- SQUARE ROOT
#-----------------------------------------------

def squareRoot (n):
	if n > 0:
		return n ** 0.5	
	if n == 0:
		return 0
	if n < 0:	
		return 'inf'
n = int(raw_input("enter a number "))
print  n**0.5		
		
#-----------------------------------------------
#        9- FACTORIAL
#-----------------------------------------------

def factorial (n):
	if n > 0:
		return n * factorial (n- 1)
	if n < 0:	
		return 'inf'
	return 1	#or return n
#-----------------------------------------------
#        10- MODULO
#-----------------------------------------------

def modulo (first,second):
	return first % second 

File: CA4_calculator.R
#-----------------------------------------------
#        1- ADDITION
#-----------------------------------------------
addition <- function (x,y) {
  return (x + y)
}

#-----------------------------------------------
#       2- Subtraction
#-----------------------------------------------
subtraction <-function (x,y){
  return (x - y)
}

#-----------------------------------------------
#       3- MULTIPLICATION
#-----------------------------------------------	
multiply <- function (x,y){
  return (x * y)	
}

#-----------------------------------------------
#        4- DIVISION
#-----------------------------------------------

divide <- function (x,y){
  if (y == 0){
    return ('inf')
  }    
  else{
    return (x / y)
  }
} 

#-----------------------------------------------
#         5- EXPONENT
#-----------------------------------------------	

exponent <- function (x,y){
  
  if (x != 0 && y ==1){
    return (x) 
  }  
  else if (x == 0 && y >0){
    return ("0")
  }  
  else if (x != 0 && y ==0){
    return ("1")
  } 
  else if (x != 0 && y <0){
    return ("For negative exponents, take the reciprocal of the base (flip it); change the negative exponent to a positive exponent and solve")
  }
  else
    return (x ^ y)
}  
#-----------------------------------------------
#         6- SQUARE
#-----------------------------------------------	

square <- function (x,y){
  if (x == 0){
    return (0)
  }  
  else if (y == 2){
    return (x ** y)	
  }
}
#-----------------------------------------------
#         7- CUBE
#-----------------------------------------------	

cube <- function (x,y){
  if (x == 0){
    return (0)
  }  
  else if (y == 3){
    return (x ** y)	
  }
}
#-----------------------------------------------
#       8- SQUARE ROOT
#-----------------------------------------------

squareRoot <- function (n){
  if (n > 0){
    return (n ** 0.5)
  }  
  else if (n == 0){
    return (0)
  } 
  else if (n < 0){	
    return ("inf")
  
  }
}
#-----------------------------------------------
#        9- FACTORIAL
#-----------------------------------------------

factor <- function (n){
  n = as.integer(readline(prompt="Enter a number: "))
  factorial = 1
  if(n < 0) {
    print("Sorry, factorial does not exist for negative numbers")
  } else if(n == 0) {
    print("The factorial of 0 is 1")
  } else {
    for(i in 1:n) {
      factorial = factorial * i
    }
    print(paste("The factorial of", n ,"is",factorial))
  }
}

#-----------------------------------------------
#        10- MODULO
#-----------------------------------------------

modulo <- function(x,y){
  return (x %% y)
}  


